soroban-sdk/src/testutils.rs

```diff
@@ -605,3 +620,75 @@ impl StellarAssetContract {
         self.asset.clone()
     }
 }
+
+/// Input for creating an [`Env`] from a custom snapshot source.
+///
+/// This struct enables [`Env::from_ledger_snapshot`] to accept custom snapshot
+/// source types beyond [`LedgerSnapshot`], providing flexibility for testing
+/// scenarios that load ledger state from different sources such as RPC endpoints,
+/// history archives, or in-memory data structures.
+///
+/// # Fields
+///
+/// * `source` - A snapshot source implementing the [`SnapshotSource`] trait.
+///   This is used to load ledger entries on demand during test execution.
```

Comment:
So, if we assume `source` pulls from an RPC, and the test being executed has multiple steps, and performs a token transfer first, does `source.get` need to do anything to fetch the updated token balance? Or is that managed by the test env itself, and `source.get` is only used for initialization of ledger entries?